### PR TITLE
SUBMARINE-367. Support submarine configuration outside k8s

### PR DESCRIPTION
**Pull request by liuxunorg**
### What is this PR for?
We need to enable the submarine running in k8s to use an external submarine-site.xml configuration file.
In this way, the user can configure the submarine-site.xml file externally, and then allow the submarine running in k8s to run according to the user's settings.

#### Test
1. Install k8s use by KIND
```
./submarine-cloud/hack/kind-cluster-build.sh
```

2. Install submarine cluster on k8s
```
./submarine-cloud/hack/deploy-submarine.sh
```
modify ./submarine-cloud/hack/conf/submarine-site.xml & log4j.properties

submarine-site.xml & log4j.properties automation mount to submarine-server pod

You can input `http://127.0.0.1` to chrome open submarine workbench web.

3. uninstall submarine cluster on k8s
```
./submarine-cloud/hack/deploy-submarine.sh -u
```

### What type of PR is it?
[Feature]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-367

### How should this be tested?
* https://travis-ci.org/liuxunorg/submarine/builds/644276787

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-cloud/hack/deploy-submarine.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    cp ${SUBMARINE_HOME}/conf/log4j.properties.template ${ROOT}/hack/conf/log4j.properties
+  fi
+
+  echo ""
+  echo -e "Have you configured the \033[31m${ROOT}/hack/conf/submarine-site.xml\033[0m file?"
+  echo -e "Have you configured the \033[31m${ROOT}/hack/conf/log4j.properties\033[0m file?"
+  echo -n "Do you want to deploy submarine in k8s cluster now? [y/n]"
+  read myselect
+  if [[ "$myselect" == "y" || "$myselect" == "Y" ]]; then
+    if kubectl get configmap --namespace default | grep submarine-config >/dev/null ; then
```
Comment (jiwq) [inline submarine-cloud/hack/deploy-submarine.sh]: Deploy submarine under the `custom namespace(submarine)` is the good practices.

Comment (liuxunorg) [reply submarine-cloud/hack/deploy-submarine.sh]: Yes, I plan to complete the functions first, and then optimize.

File: submarine-cloud/hack/deploy-submarine.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/.. && pwd)
+cd $ROOT
+SUBMARINE_HOME=${ROOT}/..
+
+source $ROOT/hack/lib.sh
+
+hack::check_requirements
+
+# install submarine in k8s cluster
```
Comment (yuanzac) [inline submarine-cloud/hack/deploy-submarine.sh]: Change install to Install

File: submarine-cloud/hack/kind-cluster-build.sh [hunk 3]
```diff
@@ -86,24 +86,7 @@ echo "k8sVersion: ${k8sVersion}"
 echo "volumeNum: ${volumeNum}"
 
 # check requirements
```
Comment (yuanzac) [inline submarine-cloud/hack/kind-cluster-build.sh]: Change check to Check

File: submarine-cloud/hack/lib.sh [hunk 4]
```diff
@@ -76,3 +76,25 @@ function hack::ensure_kind() {
 function hack::version_ge() {
     [ "$(printf '%s\n' "$1" "$2" | sort -V | head -n1)" = "$2" ]
 }
+
+function hack::check_requirements() {
+    # check requirements
```
Comment (yuanzac) [inline submarine-cloud/hack/lib.sh]: Change check to Check

**Pull request closed without merging**